### ISSUE TITLE
Custom Indexer shows error and shadows results of other indexers

### ISSUE THREAD

**Issue by mrvoss** [teaminmedias-pluswerk/ke_search#15]
I have a custom indexer that I added to $GLOBALS['TCA']['tx_kesearch_indexerconfig']['columns']['type']['config']['items'] so that I can add indexer configs for this indexer. With such a custom indexer config, when indexing, my custom indexer (now being part of \tx_kesearch_indexer::defaultIndexerTypes) is searched for in ext:ke_search/Classes/indexer/types/ and a error message shown when it is not found in this directory. The indexing continues - however, the error case throws away the results of the other indexers. This doesn't make sense - especially as a custom indexer can not be in that directory.

See also https://www.typo3-macher.de/facettierte-suche-ke-search/dokumentation/ein-eigener-indexer/ , search for "Der Fehler wäre vermutlich nicht ganz so verwirrend, wenn es eine Warnung wäre."

Suggested solution: Just remove the not found message. The indexers bundled with ke_search don't need this, and custom indexers are not responsibilty of ke_search, but of the author of the custom indexer. Patch attached.

[custom_indexer_error.patch.txt](https://github.com/teaminmedias-pluswerk/ke_search/files/81585/custom_indexer_error.patch.txt)

**Comment by fjacobi** [comment]
Why are you inserting directly into TCA and not using the recommended hooks?

**Comment by mrvoss** [comment]
I did not use the registerIndexerConfiguration() hook because it is called as itemProcFunc from ext:ke_search/Configuration/TCA/tx_kesearch_indexerconfig.php, and my understanding is that an itemProcFunc is used to only fill items. I wanted more than filling items, eg. adding custom fields to the customer indexer config, which usually is done with a TCA-override. I then reasoned that if I have to override the TCA anyway, I can do the whole TCA modification, including adding the custom indexer item, there, instead of doing one part in the hook and another in a TCA override. But this doesn't work, as mentioned above.

While changing GLOBAL TCA (eg. $GLOBALS['TCA']['tx_kesearch_indexerconfig']['columns']['sysfolder']['displayCond'] .= ',customnews';) from itemProcFunc / registerIndexerConfiguration works, it doesn't strike me as a clean way to do this (I think a TCA override would be the proper way). Similar, detecting $this->defaultIndexerTypes from $GLOBALS['TCA']['tx_kesearch_indexerconfig']['columns']['type']['config']['items'] doesn't seem to be the best way, neither (I think reading the content of ext:ke_search/Classes/indexer/types or similar would be better).

That being said, "using the recommended hooks" seems to work. So unless you see any actionable merit in my remarks, I can close this issue.
